pkg/apis/version/conversion.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SerializeToMetadata serializes the input field and adds it as an annotation to
+// the metadata under the input key.
+func SerializeToMetadata(meta *metav1.ObjectMeta, field interface{}, key string) error {
```

Comment:
- Shall we have tests for the two exported functions?
- Do we need to test with `v1.Task`?

Reply:
> * Shall we have tests for the two exported functions?

Good point, thanks! Added a test.

> * Do we need test cases with `v1.Task` in task_conversion_tests?

No; here we are ensuring that a field of v1beta1 task that is not present in v1 is still there when converted to v1 and back. There's no equivalent for v1, since it's our highest version.